src/components/TextInput/v1/TextInput.md

```diff
@@ -0,0 +1,294 @@
+```
+
+Idle Filled / Focused Filled
+
+When a field has been filled in by the user, and has been unfocused it shows idle and filled in. When a user has filled in a text input and clicked back into the text field it is shown as focused and has an icon button that allows the user to quickly clear the whole field and edit it from scratch.
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" isDark />
+  </div>
+</div>
+```
+
+Valid
+
+When information is required and/or needs to be formatted in a specific way the text input should be validated. When it’s successful the field shows a success state. Passing the `hasBeenValidated` prop to the input will enabled the valid state.
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" hasBeenValidated />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" hasBeenValidated isDark />
+  </div>
+</div>
+```
+Invalid
+
+When information is required and/or needs to be formatted in a specific way the text input should be validated. When a input can’t be validated the input is highlighted with error styling. Passing an `errors` array to the input to enabled the invalid state.
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" errors={["error"]} />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" errors={["error"]} isDark />
+  </div>
+</div>
+```
+Read Only / Disabled
+
+A disabled input is used when an action needs to be taken in before a button can be enabled. A user us unable to input information into a disabled input. Passing the `isReadOnly` prop to enabled read only / disabled state.
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" isReadOnly />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" isReadOnly isDark />
+  </div>
+</div>
+```
+###### Input Icons
+Text inputs have the ability to add a custom icon to the input to give extra content to the input. Text input icons has two available props to customize the input.
```

Comment:
Change to:

You can add a custom icon to the right side of any text input and optionally provide a click handler for it. Use these two props: